packages/back-end/src/services/experimentTimeSeries.ts

```diff
@@ -0,0 +1,294 @@
+import md5 from "md5";
+import {
+  getAllMetricIdsFromExperiment,
+  isFactMetricId,
+} from "shared/experiments";
+import { ReqContext } from "back-end/types/organization";
+import {
+  ExperimentAnalysisSummary,
+  ExperimentAnalysisSummaryResultsStatus,
+  ExperimentInterface,
+  GoalMetricStatus,
+  GuardrailMetricStatus,
+} from "back-end/src/validators/experiments";
+import {
+  ExperimentSnapshotAnalysisSettings,
+  ExperimentSnapshotInterface,
+  ExperimentSnapshotSettings,
+  MetricForSnapshot,
+  SnapshotMetric,
+} from "back-end/types/experiment-snapshot";
+import {
+  CreateMetricTimeSeriesSingleDataPoint,
+  MetricTimeSeriesValue,
+  MetricTimeSeriesVariation,
+} from "back-end/src/validators/metric-time-series";
+import {
+  FactMetricInterface,
+  FactTableInterface,
+} from "back-end/types/fact-table";
+import { getFactTableMap } from "back-end/src/models/FactTableModel";
+
+export async function updateExperimentTimeSeries({
+  context,
+  previousAnalysisSummary,
+  experiment,
+  experimentSnapshot,
+  notificationsTriggered,
+}: {
+  context: ReqContext;
+  previousAnalysisSummary?: ExperimentAnalysisSummary;
+  experiment: ExperimentInterface;
+  experimentSnapshot: ExperimentSnapshotInterface;
+  notificationsTriggered: string[];
+}) {
+  // Only update time series for dimensionless snapshots, but if we want to
+  // support dimensions for time series, we should revisit this
+  if (
+    experimentSnapshot.dimension !== null &&
+    experimentSnapshot.dimension !== ""
+  ) {
+    return;
+  }
+
+  const metricsIds = getAllMetricIdsFromExperiment(experiment);
```

Comment:
You can either get them from the snapshot like you mentioned offline, or you can keep using this. If you keep using this, you should turn off `includeActivationMetric`, and I think you need to include `metricGroups` as some IDs can be metric groups instead of individual metrics.

Reply:
Doing both -- using experimentSnapshotSettings to get the metrics, but still using the same function + passing in metricGroups + removed activation metric.